test/ctia/http/generative/properties.clj

```diff
@@ -36,45 +42,73 @@
              [spec :as fs]
              [utils :as fu]]))
 
-(defn api-for-route [model-type entity-gen]
-  (for-all
-    [new-entity entity-gen]
-    (let [app (helpers/get-current-app)
-          {:keys [get-in-config]} (helpers/get-service-map app :ConfigService)
-
-          {post-status :status
-           {id :id
-            type :type
-            :as post-entity} :parsed-body}
-          (POST app
-                (str "ctia/" (name model-type))
-                :body new-entity)]
-
-      (if (not= 201 post-status)
-        (throw (ex-info "POST did not return status 201"
-                        post-entity)))
-
-      (let [url-id
-            (-> (id/->id type id (get-http-show (app->HTTPShowServices app)))
-                :short-id
-                encode)
-
-            {get-status :status
-             get-entity :parsed-body
-             :as response}
-            (GET app
-                 (str "ctia/" type "/" url-id))]
-
-        (if (not= 200 get-status)
-          (throw (ex-info "GET did not return status 200"
-                          response)))
-
-        (if-not (empty? (keys new-entity))
-          (common= new-entity
-                   post-entity
-                   (dissoc get-entity :id))
-          (common= post-entity
-                   (dissoc get-entity :id)))))))
+(defn check-differences-in-common-key-paths
+  "Like (apply common= (vals id->m)) but
+  with more specific error messages."
+  [id->m]
+  {:pre [(seq id->m)]}
+  (let [id->m-at-path (fn [key-path]
+                        (into {}
+                              (map (fn [[id m]]
+                                     [id (get-in m key-path)]))
+                              id->m))
+        common-key-paths (->> (vals id->m)
+                              (apply keys-in-all)
+                              ;; report longest paths first
+                              (sort-by count >))]
+    (assert (seq common-key-paths))
+    (doseq [common-key-path common-key-paths
+            :let [id->vals-at-path (id->m-at-path common-key-path)]]
+      (is (apply = (vals id->vals-at-path))
+          ;; `testing` doesn't seem to be recognized by `checking`, so
+          ;; put debugging helpers here
+          (str (prn-str common-key-path)
+               (pr-str id->vals-at-path))))))
+
+(defn api-for-route
+  "Returns a function that performs"
+  [model-type entity-gen]
+  (fn this
+    ([] (this 100))
+    ([checking-options]
+     (checking
```

Comment:
This is nice. Somehow I missed the point when you introduced the use of `com.gfredericks.test.chuck.clojure-test/checking`. It looks like not the first time you're using it. I've never used that macro before. I like it. https://blog.colinwilliams.name/clojure/testing/2015/01/26/alternative-clojure-dot-test-integration-with-test-dot-check.html

Reply:
I like it too, but it's based on `with-redefs`, I hope that it will not cause future problems.

Reply:
It's great, you get the dual advantage of the shrunk test case, and the `is` error message for the shrunk case, which is often very small here.

> I like it too, but it's based on with-redefs, I hope that it will not cause future problems.

Oh? I could only find this use of `binding`. Seems like a clean abstraction at first glance.

```clojure
     (binding [ct/report #(swap! reports-atom conj %)]
       (f))
```

Reply:
Ah, the blog post talks about `with-redefs`, but they [fixed it](https://github.com/gfredericks/test.chuck/blob/master/src/com/gfredericks/test/chuck/clojure_test.cljc) to use `binding` instead.